src/aap_eda/api/serializers/activation.py

```diff
@@ -220,48 +231,130 @@ class Meta:
-            "description": activation["description"],
-            "is_enabled": activation["is_enabled"],
+            "id": activation.id,
+            "name": activation.name,
+            "description": activation.description,
+            "is_enabled": activation.is_enabled,
             "decision_environment": decision_environment,
-            "status": activation["status"],
-            "git_hash": activation["git_hash"],
+            "status": activation.status,
+            "git_hash": activation.git_hash,
             "project": project,
             "rulebook": rulebook,
             "extra_var": extra_var,
             "instances": ActivationInstanceSerializer(
-                activation["instances"], many=True
+                activation_instances, many=True
             ).data,
-            "restart_policy": activation["restart_policy"],
-            "restart_count": activation["restart_count"],
-            "rulebook_name": activation["rulebook_name"],
-            "current_job_id": activation["current_job_id"],
-            "rules_count": activation["rules_count"],
-            "rules_fired_count": activation["rules_fired_count"],
-            "created_at": activation["created_at"],
-            "modified_at": activation["modified_at"],
-            "restarted_at": activation["restarted_at"],
-            "status_message": activation["status_message"],
+            "restart_policy": activation.restart_policy,
+            "restart_count": activation.restart_count,
+            "rulebook_name": activation.rulebook_name,
+            "current_job_id": activation.current_job_id,
+            "rules_count": rules_count,
+            "rules_fired_count": rules_fired_count,
+            "created_at": activation.created_at,
+            "modified_at": activation.modified_at,
+            "restarted_at": restarted_at,
+            "status_message": activation.status_message,
         }
+
+
+class PostActivationSerializer(serializers.ModelSerializer):
+    """Serializer for the Activation model."""
+
+    name = serializers.CharField(required=True)
+    decision_environment_id = serializers.IntegerField(
+        validators=[validation.check_if_de_exists]
+    )
+
+    def validate(self, data):
+        user_id = self.initial_data["user_id"]
+        validation.check_awx_tokens(user_id)
+        extra_var_id = self.initial_data.get("extra_var_id")
+        if extra_var_id:
+            validation.check_if_extra_var_exists(extra_var_id)
+
+        return data
+
+    class Meta:
+        model = models.Activation
+        fields = [
+            "id",
+            "name",
+            "status",
+            "decision_environment_id",
+            "extra_var_id",
+            "user_id",
+            "created_at",
+            "modified_at",
+        ]
+        read_only_fields = [
+            "id",
+            "created_at",
+            "modified_at",
+        ]
+
+
+def get_rules_count(ruleset_stats: dict) -> tuple[int, int]:
```

Comment:
Not sure why the tests are passing, because you are renaming this function and removing the old `_get_rules_count` which is used here: https://github.com/ansible/eda-server/blob/bcceb144fe74dc2fa04a427745adab03a35819e4/tests/integration/api/test_activation.py#L619 but I don't see in the PR that line modified.

Reply:
Oh, I see, we are duplicating the code:
https://github.com/ansible/eda-server/blob/bcceb144fe74dc2fa04a427745adab03a35819e4/tests/integration/api/test_activation.py#L655

All of this seems quite wrong, we assert a value set by a copy of the function.
I don't say to change these tests in this PR but at least add a TODO comment.

Reply:
Removed the duplicate and added a test for `get_rules_count` function